### ISSUE TITLE
[Feature request] Ctags

### ISSUE THREAD

**Issue by kristijanhusak** [onivim/oni2#1905]
Any plans to support ctags, for both jumping and autocompletion?

**Comment by CrossR** [comment]
I think there is existing VSCode plugins for this, so it would be interesting to see/compare once we've got a more complete extension host if that is enough.

For a lot of things, we are going to need to test the VSCode extension and see, then we can make a decision for "Good/Good enough" / "We could do better as a native plugin" / "We could do better if we try and support a specific vim plugin".

**Comment by bryphe** [comment]
> I think there is existing VSCode plugins for this, so it would be interesting to see/compare once we've got a more complete extension host if that is enough.

Definitely! I was experimenting with some of the VSCode plugins - [Exuberant CTags](https://marketplace.visualstudio.com/items?itemName=chriswheeldon.exuberant-ctags) seems like it works pretty well - I set it up on [`libvim`](https://github.com/onivim/libvim) and got both definitions, and completion items. 

There were a couple of things we need to fix:
- The `ctags.excludePatterns` wasn't being picked up automatically, and is needed for the extension to function, so will need to investigate and fix that. This is a bug to be fixed on the Onivim 2 side.
- I didn't see a published `vsix` in the Releases tab, so I had to build from source and install - not a smooth flow. It also isn't available on the [open-vsx](https://open-vsx.org/), so we'll want to publish and make a `.vsix` available.
- CTags support, once we've validated the extension, will be important for us to document - it's a common workflow in the Vim world.

In addition, the [`vscode-cpptools`](https://github.com/microsoft/vscode-cpptools/) may have ctags support as well... at least this issue suggests it: https://github.com/microsoft/vscode-cpptools/issues/3900 Need to test that further.

**Comment by bryphe** [comment]
I've added the exuberant-ctags plugin, so it's installable from Onivim now:

![image](https://user-images.githubusercontent.com/13532591/90302683-c6aff800-de5c-11ea-94f1-98b6b08ec2af.png)

Provides completion:
![image](https://user-images.githubusercontent.com/13532591/90302698-e9421100-de5c-11ea-944a-fb6785d32b2a.png)

Hover:
![image](https://user-images.githubusercontent.com/13532591/90302705-f3fca600-de5c-11ea-93cd-04b839a5da3c.png)

And go-to definition

It also automatically will generate tags

More documentation here:https://onivim.github.io/docs/languages/c#ctags

Let me know if this doesn't work for you! The `clangd` extension also looks pretty nice if your environment supports it

**Comment by kristijanhusak** [comment]
Does this work with universal ctags? I wasn't able to make it work for javascript project using universal ctags.
